src/app/bundles/document-selection/document-selection.component.ts

```diff
@@ -8,11 +9,38 @@ import { MockCaseDocuments } from '../shared/mockcasedocumentdata';
 })
 export class DocumentSelectionComponent implements OnInit {
 
-  cases = MockCaseDocuments;
+  documents = MockCaseDocuments;
+  selectAllStatus: boolean;
+
+  @ViewChildren('documentItem') documentItem: QueryList<DocumentItemComponent>;
 
   constructor() { }
 
   ngOnInit() {
+    this.selectAllStatus = false;
+  }
+
+  updateSelectAll() {
+    this.selectAllStatus = !this.selectAllStatus;
+    this.documentItem.map((document: DocumentItemComponent) => document.checked = this.selectAllStatus);
   }
 
+  isAllCheckboxSelected() {
```

Comment:
it seems odd that this method starts with is and mutates, and also does not return a boolean. Just convention I guess

Reply:
I changed it to checkAllCheckboxSelected(), not sure if this is better, let me know if you think of a better method name.